hibernate-core/src/test/java/org/hibernate/test/annotations/formula/JoinFormulaWithOneToMany.java

```diff
@@ -0,0 +1,147 @@
+
+        PersistenceUnitInfo info = new PersistenceUnitInfo() {
+            @Override
+            public String getPersistenceUnitName() {
+                return "test";
+            }
+
+            @Override
+            public String getPersistenceProviderClassName() {
+                return null;
+            }
+
+            @Override
+            public PersistenceUnitTransactionType getTransactionType() {
+                return null;
+            }
+
+            @Override
+            public DataSource getJtaDataSource() {
+                return null;
+            }
+
+            @Override
+            public DataSource getNonJtaDataSource() {
+                return null;
+            }
+
+            @Override
+            public List<String> getMappingFileNames() {
+                return null;
+            }
+
+            @Override
+            public List<URL> getJarFileUrls() {
+                return null;
+            }
+
+            @Override
+            public URL getPersistenceUnitRootUrl() {
+                return null;
+            }
+
+            @Override
+            public List<String> getManagedClassNames() {
+                return Arrays.asList(Parent.class.getName(), Child.class.getName());
+            }
+
+            @Override
+            public boolean excludeUnlistedClasses() {
+                return false;
+            }
+
+            @Override
+            public SharedCacheMode getSharedCacheMode() {
+                return null;
+            }
+
+            @Override
+            public ValidationMode getValidationMode() {
+                return null;
+            }
+
+            @Override
+            public Properties getProperties() {
+                return new Properties();
+            }
+
+            @Override
+            public String getPersistenceXMLSchemaVersion() {
+                return null;
+            }
+
+            @Override
+            public ClassLoader getClassLoader() {
+                return null;
+            }
+
+            @Override
+            public void addTransformer(ClassTransformer transformer) {
+
+            }
+
+            @Override
+            public ClassLoader getNewTempClassLoader() {
+                return null;
+            }
+
+        };
+
+        new EntityManagerFactoryBuilderImpl(new PersistenceUnitInfoDescriptor(info),
+                                            Collections.singletonMap("hibernate.dialect", "org.hibernate.dialect.H2Dialect")).build();
+
+    }
+
+    @Entity @Table(name = "child") class Child {
+
+        @Id @GeneratedValue(strategy = GenerationType.AUTO) private Long id;
+
+        @ManyToOne() private Parent parent;
+
+    }
+
+    @Entity @Table(name = "parent") class Parent {
+
+        @Id @GeneratedValue(strategy = GenerationType.AUTO) private Long id;
+
+        @OneToMany(mappedBy = "parent") @JoinFormula(value = "(SELECT * FROM child)") private Set<Child> childs;
+
+    }
+
```

Comment:
From https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html#associations-JoinFormula, it seems `@JoinFormula` should be used on `Child`, not on `Parent`, for basically it is a customized foreign key.

Reply:
It has the exact same behaviour both ways. But you are right, It should be the other way so let me change the test.

Reply:
Why are you sure it has the same behaviour both ways? If it is not mentioned in the comprehensive user guide, we can assume it is not supported.

Reply:
It is not supported. But some people on the staff agree this is an interestig feature to add. Anyway, if It is not going to be done in a short term, in my opinion It should be throwing a custom exception saying this is not supported.